static/app/stores/selectedGroupStore.tsx

```diff
@@ -55,7 +55,7 @@ const storeConfig: SelectedGroupStoreDefinition = {
   },
 
   getState() {
-    return this.records;
+    return Object.fromEntries(this.records);
```

Comment:
Had to convert to an object because Map keeps the same reference on changes and thus the state never updated from `useLegacyStore`. @evanpurkhiser what do you think?